pkgs/tools/networking/maxscale/default.nix

```diff
@@ -0,0 +1,126 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, gcc, glibc
+, bison2, curl, flex, gperftools, jansson, jemalloc, kerberos, lua, mariadb
+, ncurses, openssl, pcre, pcre2, perl, rabbitmq-c, sqlite, tcl
+, libaio, libedit, libtool, libui, libuuid, zlib
+}:
+
+stdenv.mkDerivation rec {
+  name = "maxscale-${version}";
+  version = "2.1.13";
+
+  src = fetchFromGitHub {
+    owner = "mariadb-corporation";
+    repo = "MaxScale";
+    rev = "${name}";
+    sha256 = "0bwm98mbjay24vf95rk4slvjqxqblx14hvv75dpp1lhfhw25hhrf";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+
+  buildInputs = [
+    bison2 curl flex gperftools jansson jemalloc kerberos lua mariadb.connector-c
+    ncurses openssl pcre pcre2 perl rabbitmq-c sqlite tcl
+    libaio libedit libtool libui libuuid zlib
+  ];
+
+  patches = [ ./getopt.patch ];
+
+  preConfigure = ''
+    for i in `grep -l -R '#include <getopt.h>' .`; do
+      substituteInPlace $i --replace "#include <getopt.h>" "#include <${glibc.dev}/include/getopt.h>"
```

Comment:
For a properly configured toolchain, it seems odd to have to substitute in a full path to the libc headers; what motivated this?

Reply:
Without this patch, getopt.h provided with maxscale, was included instead of glibc getopt.h, and that caused weird errors, like:

`
[ 31%] Building CXX object server/core/CMakeFiles/maxscale.dir/gateway.cc.o
In file included from /nix/store/p2bkk31685y6b4l30bp9961g6kd133wd-glibc-2.26-115-dev/include/bits/getopt_posix.h:27:0,
                 from /nix/store/p2bkk31685y6b4l30bp9961g6kd133wd-glibc-2.26-115-dev/include/unistd.h:872,
                 from /tmp/nix-build-maxscale-2.1.13.drv-0/source/server/core/gateway.cc:47:
/nix/store/p2bkk31685y6b4l30bp9961g6kd133wd-glibc-2.26-115-dev/include/bits/getopt_core.h:91:12: error: declaration of 'int getopt(int, char* const*, const char*) throw ()' has a different exception specifier
 extern int getopt (int ___argc, char *const *___argv, const char *__shortopts)
                 ^~~~~~
In file included from /tmp/nix-build-maxscale-2.1.13.drv-0/source/server/core/gateway.cc:42:0:
/nix/store/b1mqlzdnp32bzhbmqsk99hjmkk19rdrq-mariadb-connector-c-2.3.4/include/mariadb/getopt.h:108:12: note: from previous declaration 'int getopt(int, char* const*, const char*)'
 extern int getopt (int argc, char *const *argv, const char *shortopts);
                 ^~~~~~
make[2]: *** [server/core/CMakeFiles/maxscale.dir/build.make:63: server/core/CMakeFiles/maxscale.dir/gateway.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:596: server/core/CMakeFiles/maxscale.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
builder for ‘/nix/store/x1if3dad7j0i24lckdz4jbyls9ifa6jg-maxscale-2.1.13.drv’ failed with exit code 2
error: build of ‘/nix/store/x1if3dad7j0i24lckdz4jbyls9ifa6jg-maxscale-2.1.13.drv’ failed

`

Reply:
Ugh ... why would they do that?